### ISSUE TITLE
Make `ask` and `askargs` API return a value for `format=count`

### ISSUE THREAD

**Issue by osnard** [SemanticMediaWiki/SemanticMediaWiki#4857]
Currently when running a query with the Action API modules `ask` or `askargs` no value will be returned.

E.g. `api.php?action=askargs&conditions=Modification%20date::%2B&printouts=Modification%20date&parameters=sort%3DModification%20date|`**`format%3Dcount`**`&format=json` will return 

```
{
	"query": {
		"printrequests": [...],
		"results": [],
		"serializer": "SMW\\Serializers\\QueryResultSerializer",
		"version": 2,
		"meta": {...}
	}
}
```
The query is executed correctly and the `SMWQueryResult` object contains the proper value. it is just not returned. it would be nice to have Action API modules `ask` and `askargs` respect  `format=count`

**Comment by kghbln** [comment]
Reopening this. See also #4890

**Comment by akuckartz** [comment]
Can this be closed? #4890 was closed.

**Comment by kghbln** [comment]
> Can this be closed? #4890 was closed.

No, since this is feature request. #4890 was a bug as a result of an unsuccessful implementation attempt

**Comment by D-Groenewegen** [comment]
**+1** 
I just ran into this. The ask API doesn't support result formats but I would welcome an additional parameter to the ask API, maybe `totalcount`, along with the other data in `meta`. 

If I understand correctly, the only way you can retrieve the total number of results from the API is :
- use the ask API with limit set to infinite (well, 99999999999999999) and read `count` . Cons: painfully slow
- use askarg API with `format=count`: feasible only for rather basic queries like `<some property>::<some value>`
- use MW's parser API on full query syntax with `format=count` : circuitous and potentially slow
- use SMW with ElasticSearch and use its search API : available only to the lucky few

**Comment by gesinn-it-gea** [comment]
Fixed in commit [2be6e6757](https://github.com/SemanticMediaWiki/SemanticMediaWiki/commit/2be6e6757) (see also #7006).

A previous attempt (PR #4858, commit c9b7a6ff, reverted after #4890) placed the count check inside `getResult()`, which caused `countValue` to bleed into normal queries and broke existing API clients.

The new fix intercepts in `doFormat()` instead, guarded by `getCountValue() !== null`. Since `setCountValue()` is only called by the query engine when running in `MODE_COUNT`, this guard never fires for normal queries — making the change fully non-breaking.

The API now returns:
```json
{
  "query": {
    "count": 42,
    "meta": { "type": "count" }
  }
}
```
for queries like `[[Category:Person]]|format=count`.
